File: Code/Aula_07/Ex5.c
#include <stdio.h>
#include <stdlib.h>
#include <signal.h>

void funcao_sigsegv()
{
	printf("Recebi segment fault. Vou morrer!!!\n");   
	exit(1);
}

int main()
{
	char *p;
	signal(SIGSEGV, funcao_sigsegv);
	printf("Vou forcar um segment fault.\n");
	printf("%s", *p);
}

File: 2_Programação_sistemas_operacionais/2.4_Pipes_Sinais_Alarmes/Python/Ex5.py
import signal, sys

def funcao_sigsegv(signum, frame):
	print("Recebi segment fault. Vou morrer!!!")
	exit(1)

def main():
	main()

signal.signal(signal.SIGSEGV, funcao_sigsegv)

print("Vou forçar um segment fault.")

sys.setrecursionlimit(50000)
main()